scale-tests/run.sh

```diff
@@ -132,12 +209,25 @@ if is_interactive; then
   done
 fi
 
-if docker inspect -f {{.State.Running}} "${CONTAINER_NAME}" > /dev/null 2>&1; then
-  log "Container '${CONTAINER_NAME}' already running, skipping Docker image build"
-else
-  git clone git@github.com:mesosphere/spark-build.git "${TEST_DIRECTORY}" | tee -a "${LOG_FILE}"
+set +e
+docker inspect -f {{.State.Running}} "${CONTAINER_NAME}" > /dev/null 2>&1
+readonly container_running=$?
+
+docker exec -it "${CONTAINER_NAME}" test -f "${CONTAINER_FINISHED_SETTING_UP_FILE}"
+readonly container_finished_setting_up=$?
+set -e
 
-  docker build -t "${IMAGE_NAME}" "${TEST_DIRECTORY}/scale-tests" | tee -a "${LOG_FILE}"
+if [ ${container_running} -ne 0 ] || [ ${container_finished_setting_up} -ne 0 ]; then
+  log "Building Docker image for ${TEST_NAME}"
+
+  log "Cleaning up possibly pre-existing containers"
+  docker kill "${CONTAINER_NAME}" > /dev/null 2>&1 || true
+  docker rm -f "${CONTAINER_NAME}" > /dev/null 2>&1 || true
```

Comment:
Nice, I had been doing this myself and telling others to do it too whenever issues popped up in the script.